File: _java/symulacja_organizmow/src/Symulacja/Zwierzeta/Wilk.java
package Symulacja.Zwierzeta;
import Symulacja.*;

import java.awt.*;

public class Wilk extends Zwierze {
        protected void SetInicjatywa(){
            this.inicjatywa=5;
        };
        protected void SetSila(){
            this.sila=9;
        };
        public Wilk(Swiat swiat){
            super(swiat);
            SetInicjatywa();
            SetSila();
        };
        public char Rysowanie(){
            return 'W';
        };
        public Organizm Urodz() {
            return new Wilk(this.swiat);
        };
        public boolean MozeBycPartnerem(Organizm organizm){
            if(organizm instanceof Zwierze)
                return true;
            else
                return false;
        };
    @Override
    public Color GetColor(){
        return new Color(100,100,0);
    };
}

File: _c++/symulacja_organizmow/Organizmy/Zwierzeta/Wilk.cpp
//
//  Wilk.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Wilk.hpp"

Wilk::Wilk(Swiat &swiat):Zwierze(swiat){
    SetInicjatywa();
    SetSila();
};
char Wilk::Rysowanie() const{
    return 'W';
}
void Wilk::SetInicjatywa(){
    this->inicjatywa=5;
}
void Wilk::SetSila(){
    this->sila=9;
}
Organizm* Wilk::Urodz(){
    return new Wilk(this->swiat);
}
bool Wilk::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Wilk*>(&organizm))
       return true;
    else
       return false;
}
